src/js/media-controller.js

```diff
@@ -580,62 +580,14 @@ class MediaController extends MediaContainer {
       this._pipUnavailable,
     );
 
-    // TODO: Update to propagate all states when registered
     if (this.media) {
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CAPTIONS_LIST,
-        stringifyTextTrackList(getCaptionTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SUBTITLES_LIST,
-        stringifyTextTrackList(getSubtitleTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
-        stringifyTextTrackList(getShowingCaptionTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
-        stringifyTextTrackList(getShowingSubtitleTracks(this)) || undefined
-      );
-      propagateMediaState([el], MediaUIAttributes.MEDIA_PAUSED, getPaused(this));
-      // propagateMediaState([el], MediaUIAttributes.MEDIA_VOLUME_LEVEL, level);
-      propagateMediaState([el], MediaUIAttributes.MEDIA_MUTED, getMuted(this));
-      propagateMediaState([el], MediaUIAttributes.MEDIA_VOLUME, getVolume(this));
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_VOLUME_LEVEL,
-        getVolumeLevel(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_IS_FULLSCREEN,
-        this.hasAttribute(MediaUIAttributes.MEDIA_IS_FULLSCREEN)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CURRENT_TIME,
-        getCurrentTime(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_DURATION,
-        getDuration(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SEEKABLE,
-        getSeekable(this).join(':')
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_PLAYBACK_RATE,
-        getPlaybackRate(this)
-      );
+      Object.keys(MediaUIAttributes).forEach((attribute) => {
+        propagateMediaState(
+          [el],
+          MediaUIAttributes[attribute],
+          Delegates[attribute] ? Delegeates[attribute](this) : Delegates.default(this, MediaUIAttributes[attribute])
```

Comment:
`Delegeates` typo, I think this repo really needs eslint. would help a bit

Reply:
oops. Yeah, linting would be helpful to catch silly mistakes like this.